doc/_ext/collapsible.py

```diff
@@ -0,0 +1,56 @@
+import random
+import string
+
+from docutils import nodes
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst.directives import unchanged
+
+
+class CollapsibleNode(nodes.General, nodes.Element):
+    pass
+
+
+def visit_collapsible_node(self, node):
+    self.body.append(f"""
+<input type="checkbox" id="{node["id"]}" class="hide">
+<label for="{node["id"]}">
+    <span class="sectionLbl"><b>{node["header"]}</b></span>
+</label>
+<div>
+""")
+
+
+def depart_collapsible_node(self, node):
+    self.body.append("""
+</div>
+<span class="nl">
```

Comment:
I think you can totally remove nl and sectionLbl if you change the label display mode to "block".